src/Pages/GlobalConfigurations/Authorization/shared/components/userGroups/UserGroup.tsx

```diff
@@ -0,0 +1,985 @@
+            const sortedApplications =
+                openMenu === 'entityName/apps' || openMenu === 'entityName/jobs'
+                    ? applications
+                    : sortBySelected(permission.entityName, applications, 'value')
+            return sortedApplications
+        })
+    }, [openMenu, permission.environment, permission.entityName, projectId])
+
+    const setWorkflowsForJobs = async (perimssion) => {
+        if (abortControllerRef.current) {
+            abortControllerRef.current.abort()
+            abortControllerRef.current = null
+        }
+        abortControllerRef.current = new AbortController()
+        setWorkflowList({ loading: true, options: [] })
+        try {
+            setWorkflowList({ loading: true, options: [] })
+            const jobNames = perimssion.entityName.filter((option) => option.value != '*').map((app) => app.label)
+            const {
+                result: { appIdWorkflowNamesMapping },
+            } = await getAllWorkflowsForAppNames(jobNames, abortControllerRef.current.signal)
+            const workflowOptions = []
+            for (const jobName in appIdWorkflowNamesMapping) {
+                workflowOptions.push({
+                    label: jobName,
+                    options: appIdWorkflowNamesMapping[jobName].map((workflow) => ({
+                        label: workflow,
+                        value: workflow,
+                    })),
+                })
+            }
+            abortControllerRef.current = null
+            setWorkflowList({ loading: false, options: workflowOptions })
+        } catch (err: any) {
+            if (err.errors && err.errors[0].code != 0) {
+                showError(err)
+            }
+            setWorkflowList({ loading: false, options: [] })
+        }
+    }
+
+    function formatOptionLabel({ value, label }) {
+        return (
+            <div className="flex left column">
+                <span>
+                    {
+                        (permission.accessType === ACCESS_TYPE_MAP.HELM_APPS
+                            ? customRoles.possibleRolesMetaForHelm
+                            : permission.accessType === ACCESS_TYPE_MAP.JOBS
+                            ? customRoles.possibleRolesMetaForJob
+                            : customRoles.possibleRolesMeta)[value]?.value
+                    }
+                </span>
+                <small className="light-color">
+                    {
+                        (permission.accessType === ACCESS_TYPE_MAP.HELM_APPS
+                            ? customRoles.possibleRolesMetaForHelm
+                            : permission.accessType === ACCESS_TYPE_MAP.JOBS
+                            ? customRoles.possibleRolesMetaForJob
+                            : customRoles.possibleRolesMeta)[value]?.description
+                    }
+                </small>
+            </div>
+        )
+    }
+
+    function formatOptionLabelClusterEnv(option, { inputValue }) {
```

Comment:
Move these to utils file

Reply:
Will move this while refactoring Direct Permission component in v3